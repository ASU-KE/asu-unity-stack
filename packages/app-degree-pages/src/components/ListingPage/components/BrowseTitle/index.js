// @ts-check

import React from "react";

const BrowseTitle = () => (
  <div className="container">
    <h2>
      <span className="highlight-gold">Browse degrees</span>
    </h2>
  </div>
);

export { BrowseTitle };
