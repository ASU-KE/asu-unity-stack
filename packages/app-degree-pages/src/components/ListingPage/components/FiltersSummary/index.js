// @ts-check

import PropTypes from "prop-types";
import React from "react";

import { filterValueShape } from "../../../../core/models";
import { idGenerator } from "../../../../core/utils";

/**
 * @param {{
 *  value: import("../Filters").FiltersState
 *  onRemove: (filterName: string, filter: import("../Filters").FilterOption) => void
 * }} propsÃŽ
 * @returns {JSX.Element}
 */
function FiltersSummary({
  value: { isActive, locations, asuLocals, acceleratedConcurrent },
  onRemove,
}) {
  const genFilterId = idGenerator("filter-");
  const filterCount = locations.length + asuLocals.length;

  const SummaryItem = (filterName, filter) => (
    <span
      key={genFilterId.next().value}
      tabIndex={-1}
      role="button"
      aria-label={`Remove filter ${filter.text}`}
      data-filter-id={filterName}
      onKeyDown={() => onRemove(filterName, filter)}
      onClick={() => onRemove(filterName, filter)}
      className="btn btn-tag btn-tag-alt-white mr-2"
    >
      <span>{filter.text}</span>
      <i className="fas fa-times pl-1" />
    </span>
  );

  return (
    <div>
      <header>
        <strong>Applied filters</strong>
      </header>
      <div>
        {!isActive || filterCount === 0 ? (
          <span>No filters applied</span>
        ) : (
          <>
            {locations.map(loc => SummaryItem("locations", loc))}
            {asuLocals.map(loc => SummaryItem("asuLocals", loc))}
            {(acceleratedConcurrent.value !== "all"
              ? [acceleratedConcurrent]
              : []
            ).map(acValue => SummaryItem("acceleratedConcurrent", acValue))}
          </>
        )}
      </div>
    </div>
  );
}

FiltersSummary.propTypes = {
  value: filterValueShape,
  onRemove: PropTypes.func,
};

export { FiltersSummary };
