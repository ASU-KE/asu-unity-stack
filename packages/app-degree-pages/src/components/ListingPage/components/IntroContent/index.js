/* eslint-disable no-console */
// @ts-check
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import {
  ApplyNow,
  OverlapContentImage,
  Video,
} from "../../../../core/components";
import {
  contentPropType,
  contentPropTypes,
  imagePropType,
  tagHeadingList,
  videoPropType,
} from "../../../../core/models";
import { spreadClasses, parseHeading } from "../../../../core/utils";

/**
 * @typedef {import('../../../../core/models/listing-page-types').IntroContentProps} IntroContentProps
 * @typedef {import('../../../../core/models/shared-types').AppTitleProps} AppTitleProps
 */

const PhotoGrid = styled.div`
  .photo-item {
    padding-top: 1.25rem;
  }
`;

const SectionWrapper = styled.div`
  .uds-img,
  .uds-video-container {
    margin-top: 1.25rem;
  }
`;

/**
 * @param {IntroContentProps} props
 * @returns {JSX.Element}
 */
function textPhotoGridTypeHtmlTemplate({
  applyNowUrl,
  title,
  contents,
  photoGrid,
}) {
  const Heading = parseHeading(title);

  return (
    <section className="container">
      <div className="row pt-3">
        <div className="col col-sm-12 col-md-5 pl-0">
          <Heading>
            <span className={`${spreadClasses(title.cssClass)}`}>
              {title.text}
            </span>
          </Heading>
          {contents.map((content, index) => (
            <p
              key={`content-${index + 1}`}
              className={`${spreadClasses(content.cssClass)}`}
            >
              {content.text}
            </p>
          ))}
          <ApplyNow href={applyNowUrl} />
        </div>
        <div className="col col-sm-12 col-md-7">
          <PhotoGrid className="row">
            {photoGrid.images.map((image, index) => (
              <div
                className="photo-item col col-sm-12 col-md-6 col-lg-6"
                key={`img-${index + 1}`}
              >
                <img
                  className={`mg-fluid ${spreadClasses(image.cssClass)}`}
                  src={image.url}
                  alt={image.altText}
                  style={{ maxWidth: "100%" }}
                />
              </div>
            ))}
          </PhotoGrid>
        </div>
      </div>
    </section>
  );
}

/**
 * @param {IntroContentProps} props
 * @returns {JSX.Element}
 */
function textTextMediaTypeHtmlTemplate({
  applyNowUrl,
  title,
  contents,
  image,
  video,
}) {
  const Heading = parseHeading(title);

  return (
    <SectionWrapper className="container">
      <div className="row pt-3">
        <div className="col col-sm-12 col-md-5">
          <Heading>
            <span className={`${spreadClasses(title.cssClass)}`}>
              {title.text}
            </span>
          </Heading>
          {contents.map((content, index) => (
            <p
              key={`content-${index + 1}`}
              className={`${spreadClasses(content.cssClass)}`}
            >
              {content.text}
            </p>
          ))}
          <ApplyNow href={applyNowUrl} />
        </div>
        <div className="col col-sm-12 col-md-7">
          {video && (
            <Video
              url={video.url}
              vttUrl={video.vttUrl}
              altText={video.altText}
            />
          )}
          {image && (
            <div className="uds-img">
              <img src={image.url} className="img-fluid" alt={image.altText} />
            </div>
          )}
        </div>
      </div>
    </SectionWrapper>
  );
}

/**
 * @param {IntroContentProps} props
 * @returns {JSX.Element}
 */
function textImageOverlapTypeHtmlTemplate({ title, contents, image }) {
  return (
    <section className="container">
      <div className="row">
        <div className="col-md-12">
          <OverlapContentImage
            title={title.text}
            contents={contents}
            image={image}
            contentDirection="right"
            // contentChildren={<ApplyNow href={applyNowUrl} />}
          />
        </div>
      </div>
    </section>
  );
}

/**
 * @param {IntroContentProps} props
 * @returns {JSX.Element}
 */
const textTypeHtmlTemplate = ({ applyNowUrl, title, contents }) => {
  const Heading = parseHeading(title);

  return (
    <section className="container">
      <div className="row pt-3">
        <div className="col col-sm-12 col-md-5">
          <Heading>
            <span className={`${spreadClasses(title.cssClass)}`}>
              {title.text}
            </span>
          </Heading>
        </div>
        <div className="col col-sm-12 col-md-7 pt-2">
          {contents.map((content, index) => (
            <p
              key={`content-${index + 1}`}
              className={`${spreadClasses(content.cssClass)}`}
            >
              {content.text}
            </p>
          ))}
          <ApplyNow href={applyNowUrl} />
        </div>
      </div>
    </section>
  );
};

/**
 *
 * @param {IntroContentProps} props
 * @returns {JSX.Element}
 */
const IntroContent = ({
  applyNowUrl,
  type,
  title,
  contents,
  image,
  video,
  photoGrid,
}) => {
  const templateTypes = {
    "text": () => textTypeHtmlTemplate({ applyNowUrl, title, contents }),
    "text-media": () =>
      textTextMediaTypeHtmlTemplate({
        applyNowUrl,
        title,
        contents,
        image,
        video,
      }),
    "text-image-overlay": () =>
      textImageOverlapTypeHtmlTemplate({ applyNowUrl, title, contents, image }),
    "text-photo-grid": () =>
      textPhotoGridTypeHtmlTemplate({
        applyNowUrl,
        title,
        contents,
        image,
        photoGrid,
      }),
    "undefined": () => {
      console.error(
        `the type '${type}' is not supported by the 'IntroContent' component.`
      );
      return null;
    },
  };

  return templateTypes[type]();
};

IntroContent.propTypes = {
  applyNowUrl: PropTypes.string,
  type: PropTypes.oneOf([
    "text",
    "text-media",
    "text-image-overlay",
    "text-photo-grid",
  ]).isRequired,
  video: videoPropType,
  image: imagePropType,
  header: contentPropType,
  title: PropTypes.shape({
    ...contentPropTypes,
    component: PropTypes.oneOf(tagHeadingList),
  }),
  contents: PropTypes.arrayOf(contentPropType),
  photoGrid: PropTypes.shape({
    images: PropTypes.arrayOf(imagePropType),
  }),
};

export { IntroContent };
