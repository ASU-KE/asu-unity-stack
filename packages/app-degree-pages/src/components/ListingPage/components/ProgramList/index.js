// @ts-check
import { Pagination } from "@asu-design-system/components-core";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";

import { GRID_VIEW_ID, LIST_VIEW_ID } from "../../../../core/models";
import { computePages } from "../../../../core/utils";
import { GridView } from "./GridView";
import { ListView } from "./ListView";
import { degreeListPropTypes } from "./programs-prop-types";

/**
 *  @typedef {{
 *    programms: Object[]
 *    loading: boolean
 * }} GridListingProps
 */

/**
 * @type {Object.<string, ListView |  GridView>}
 */
const programViewer = {
  [GRID_VIEW_ID]: GridView,
  [LIST_VIEW_ID]: ListView,
};

/**
 *
 * @param {{
 *  dataViewComponent: GRID_VIEW_ID | LIST_VIEW_ID
 * } & GridListingProps} props
 */
function ProgramList({ dataViewComponent, loading, programms }) {
  const ROW_PAGES = 8;
  const TOTAL_PAGES = computePages(programms.length, ROW_PAGES);
  const ProgramsViewer = programViewer[dataViewComponent];
  const [tableView, setTableView] = useState([]);

  const onPageChange = (_, newPage) => {
    const fromRecord = (newPage - 1) * ROW_PAGES;
    const toRecord = fromRecord + ROW_PAGES;

    setTableView(programms.slice(fromRecord, toRecord));
  };

  useEffect(() => {
    setTableView(programms.slice(0, ROW_PAGES));
    // ReactTooltip.rebuild();
  }, [programms]);

  return (
    <>
      <ProgramsViewer loading={loading} programms={tableView} />

      <Pagination
        // count={5}
        type="default"
        background="white"
        totalPages={TOTAL_PAGES}
        onChange={onPageChange}
        showFirstButton
        showLastButton
      />
    </>
  );
}

ProgramList.propTypes = {
  dataViewComponent: PropTypes.string,
  ...degreeListPropTypes,
};

export { ProgramList, GridView, ListView };
