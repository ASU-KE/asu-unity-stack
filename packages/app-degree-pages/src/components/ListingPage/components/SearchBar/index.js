// @ts-check
import { Button } from "@asu-design-system/components-core";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

const Section = styled.div`
  label[for="search-field"] {
    margin-bottom: 0;
  }
`;

/**
 * @typedef {{
 *  onSearch: () => void
 *  onChange: (keyword: string) => void
 *  value: string
 * }} SearchBarProps
 */

/**
 *
 * @param {SearchBarProps} props
 * @returns {JSX.Element}
 */

const SearchBar = ({ value, onChange, onSearch }) => {
  return (
    <Section className="container mt-5">
      <form
        className="uds-form p-0 col-md-6 col-sm-12"
        onSubmit={e => {
          e.preventDefault();
          onSearch();
        }}
      >
        <div className="form-group mb-0 mr-2">
          <label htmlFor="search-field">Search</label>

          <div className="d-flex row align-items-baseline g-3">
            <div className="col-sm-12 col-md-6 align-self-end">
              <input
                id="search-field"
                value={value}
                type="text"
                className="form-control"
                placeholder="Search degree programs"
                onChange={e => onChange(e.target.value)}
              />
            </div>

            <div className="col-sm-12 col-md-6">
              <Button
                color="maroon"
                label="Search now"
                ariaLabel="Search now"
                size="default"
                onClick={onSearch}
              />
            </div>
          </div>
        </div>
      </form>
    </Section>
  );
};

SearchBar.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  onSearch: PropTypes.func,
};

export { SearchBar };
