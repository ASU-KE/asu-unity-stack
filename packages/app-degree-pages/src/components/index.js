import { DegreePage } from "./DegreePage";

export { DegreePage };
