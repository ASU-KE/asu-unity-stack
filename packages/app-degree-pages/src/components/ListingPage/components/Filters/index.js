// @ts-check
import { Button } from "@asu-design-system/components-core";
import PropTypes from "prop-types";
import React from "react";

import {
  locationOptions,
  asuLocalOptions,
  acceleratedConcurrentOptions,
} from "../../../../core/models/filters-options";
import { SelectFormGroup } from "./components";
import { Section, ButtonLink } from "./index.style";

/**
 *  @typedef  {{
 *    locations?: string [],
 *    asuLocals?: string [],
 *    acceleratedConcurrent: "acceleratedConcurrent" | "concurrentDegrees" | "all"
 * }} FiltersState
 */

/**
 *
 * @typedef {{
 *  value: FiltersState
 *  onValueChange: () => void
 *  onApplyFilters: (
 *    data: {
 *      location: string[],
 *      acceleratedConcurrent: string
 *    }
 * ) => void
 *  onCleanFilters: () => void
 * }} FilterProps
 */

/**
 *
 * @param {FilterProps} props
 * @returns {JSX.Element}
 */

const INITIAL_STATE = {
  locations: [],
  asuLocals: [],
  acceleratedConcurrent: acceleratedConcurrentOptions[0].value,
};

const Filters = ({ value, onValueChange, onApplyFilters, onCleanFilters }) => {
  const handleChangeMultipleField = (id, event) => {
    const stateMap = {
      asuLocal: () => {
        const locations = value.locations.includes("ONLNE")
          ? [...value.locations]
          : ["ONLNE", ...value.locations];

        const asuLocals = Array.from(
          event.target.selectedOptions,
          item => item.value
        );

        onValueChange({
          ...value,
          locations,
          asuLocals,
        });
      },
      location: () => {
        const locations = value.asuLocals[0]
          ? [
              "ONLNE",
              ...Array.from(event.target.selectedOptions, item =>
                item.value !== "ONLNE" ? item.value : ""
              ),
            ]
          : Array.from(event.target.selectedOptions, item => item.value);

        onValueChange({
          ...value,
          locations,
        });
      },
    };
    const renderedValue = () =>
      stateMap[id]
        ? stateMap[id]()
        : () =>
            onValueChange({
              ...value,
              [id]: Array.from(
                event.target.selectedOptions,
                item => item.value
              ),
            });

    renderedValue();
  };

  const handleChangeField = (id, event) => {
    onValueChange({ ...value, [id]: event.target.value });
  };

  const handleApplyFilters = () => {
    const { asuLocals, ...filters } = value;
    onApplyFilters?.(filters);
  };

  const handleCleanFilters = () => {
    onValueChange(INITIAL_STATE);
    onCleanFilters?.();
  };

  return (
    <Section className="container mt-4 pb-6 mb-6">
      <h4>Filter your results</h4>
      <form className="mt-3 uds-form row">
        <div className="col-lg-4 col-md-12">
          <SelectFormGroup
            multiple
            id="location"
            label="Location or online"
            selected={value.locations}
            options={locationOptions}
            onChange={handleChangeMultipleField}
          />
        </div>
        <div className="col-lg-4 col-md-12">
          <SelectFormGroup
            multiple
            id="asuLocal"
            label="ASU Local"
            selected={value.asuLocals}
            options={asuLocalOptions}
            onChange={handleChangeMultipleField}
          />
        </div>
        <div className="col-lg-4 col-md-12">
          <SelectFormGroup
            id="acceleratedConcurrent"
            label="Accelerated/Concurrent"
            selected={value.acceleratedConcurrent}
            options={acceleratedConcurrentOptions}
            onChange={handleChangeField}
          />
        </div>
      </form>
      <div>
        <Button
          color="maroon"
          label="Apply filters"
          ariaLabel="Apply filters"
          size="default"
          onClick={handleApplyFilters}
        >
          Apply filters
        </Button>
        <ButtonLink className="btn btn-link" onClick={handleCleanFilters}>
          Clean filters
        </ButtonLink>
      </div>
    </Section>
  );
};

Filters.propTypes = {
  value: PropTypes.shape({
    locations: PropTypes.arrayOf(PropTypes.string),
    asuLocals: PropTypes.arrayOf(PropTypes.string),
    acceleratedConcurrent: PropTypes.string,
  }),
  onValueChange: PropTypes.func,
  onApplyFilters: PropTypes.func,
  onCleanFilters: PropTypes.func,
};

export { Filters };
