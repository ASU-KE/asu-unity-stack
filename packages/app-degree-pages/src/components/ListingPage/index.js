// @ts-check
import { Hero } from "@asu-design-system/components-core";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";
import styled from "styled-components";

import { Loader } from "../../core/components";
import { useFetch } from "../../core/hooks/use-fetch";
import { acceleratedConcurrentValues, LIST_VIEW_ID } from "../../core/models";
import {
  columSettingsPropType,
  dataSourcePropType,
} from "../../core/models/app-prop-types";
import { degreeDataPropResolverService } from "../../core/services";
import { urlResolver } from "../../core/utils/data-path-resolver";
import { BrowseTitle } from "./components/BrowseTitle";
import { Filters } from "./components/Filters";
import { FiltersSummary } from "./components/FiltersSummary";
import { IntroContent } from "./components/IntroContent";
import { ProgramList } from "./components/ProgramList";
import { SearchBar } from "./components/SearchBar";

/**
 * @typedef {import('../../core/models/listing-page-types').ListingPageProps} ListingPageProps
 * @typedef {import("./components/Filters").FiltersState} FiltersState
 */

const Main = styled.main`
  @media (max-width: 768px) {
    & {
      font-size: 0.9rem;
    }
  }
`;

/**
 *
 * @param {ListingPageProps} props
 * @returns {JSX.Element}
 */
const ListingPage = ({
  actionUrls,
  hasSearchBar = true,
  hasFilters = true,
  hero,
  introContent,
  programList,
}) => {
  /** @type {import("../../core/hooks/use-fetch").UseFetchTuple<{programs: []}>} */
  const [{ data, loading, error }, doFetchPrograms] = useFetch();
  const [searchLoading, setSearchLoading] = useState(false);
  const [tableView, setTableView] = useState([]);
  /** @type {import("../../core/models/shared-types").UseStateTuple<LIST_VIEW_ID>} */
  const [dataViewComponent] = useState(LIST_VIEW_ID);
  // const [dataViewComponent, setDataViewComponent] = useState(LIST_VIEW_ID);
  const url = urlResolver(programList.dataSource);

  /** @type {import("../../core/models/shared-types").UseStateTuple<FiltersState>} */
  const [stateFilters, setStateFilters] = useState({
    locations: [],
    asuLocals: [],
    acceleratedConcurrent: "all",
  });
  /** @type {import("../../core/models/shared-types").UseStateTuple<string[]>} */
  const [appliedFilters, setAppliedFilters] = useState([]);

  useEffect(() => {
    doFetchPrograms(url);
  }, [url]);

  useEffect(() => {
    setTableView(data?.programs || []);
  }, [data]);

  /**
   * @param {import("src/components/ListingPage/components/Filters").FiltersState} data
   */
  const onDegreeApplyFilters = async ({ acceleratedConcurrent, locations }) => {
    // ============================================================
    // prevent search
    // ============================================================
    if (loading || searchLoading) return;

    if (!acceleratedConcurrent && locations.length === 0) {
      return;
    }
    // ============================================================

    setSearchLoading(true);

    await doFetchPrograms(url);

    /** @param {Object.<string, []>} row  */
    const isValidCampus = (row = {}) => {
      const resolver = degreeDataPropResolverService(row);

      return locations.length > 0
        ? resolver.getCampusList()?.some(campus => locations.includes(campus))
        : true;
    };

    /** @param {Object.<string, []>} row  */
    const isValidAcceleratedConcurrent = (row = {}) => {
      return acceleratedConcurrent && acceleratedConcurrent !== "all"
        ? row[acceleratedConcurrent]?.length > 0
        : true;
    };

    /** @param {Object.<string, any>} row  */
    const doFilter = row =>
      isValidCampus(row) && isValidAcceleratedConcurrent(row);
    // filter program list
    setTableView(data.programs.filter(doFilter));

    setSearchLoading(false);

    // set summary filters
    const newAppliedFilterList = [...locations];
    if (acceleratedConcurrent !== "all")
      newAppliedFilterList.push(acceleratedConcurrent.replace("Degrees", ""));

    setAppliedFilters(newAppliedFilterList);
  };

  const onDegreeCleanFilters = () => setTableView(data?.programs || []);

  /**
   *
   * @param {string} keyword
   */
  const onDegreeSearch = async (keyword = "") => {
    /* todo: this won't work since the only way to clear up
              the search is to provide an empty keyword */
    // if (!keyword.trim()) {
    //   alert("Please provide a valid keyword");
    //   return;
    // }

    setSearchLoading(true);
    await doFetchPrograms(url);
    if (data?.programs?.length > 0) {
      setTableView(
        data.programs.filter(row => {
          const resolver = degreeDataPropResolverService(row);
          return resolver.getDescrLongExtented()?.includes?.(keyword);
        })
      );
    }
    setSearchLoading(false);
  };

  const onRemoveFilterValue = filterValue => {
    // clean up: summary filters component
    const newFilters = appliedFilters.filter(f => f !== filterValue);
    setAppliedFilters(newFilters);
    // clean up: state filters component
    /** @type {FiltersState} */
    const cleanFilters = {};

    if (filterValue === "ONLNE") {
      cleanFilters["asuLocals"] = [];
      cleanFilters["locations"] = stateFilters.locations.filter(
        f => f !== filterValue
      );
    } else if (acceleratedConcurrentValues.includes(`${filterValue}Degrees`)) {
      cleanFilters["acceleratedConcurrent"] = "all";
    } else {
      cleanFilters["locations"] = stateFilters.locations.filter(
        f => f !== filterValue
      );
    }

    const newStateFilters = {
      ...stateFilters,
      ...cleanFilters,
    };

    setStateFilters({ ...newStateFilters });

    // clean up: redo query with new filters
    onDegreeApplyFilters({
      acceleratedConcurrent: newStateFilters.acceleratedConcurrent,
      locations: newStateFilters.locations,
    });
  };

  return (
    <>
      {hero ? (
        <Hero
          image={hero.image}
          title={{ ...hero.title, maxWidth: "100%" }}
          contents={hero.contents}
        />
      ) : null}

      <Main className="container" data-is-loading={loading}>
        {introContent ? (
          <IntroContent
            applyNowUrl={actionUrls?.applyNowUrl}
            type={introContent.type}
            header={introContent.header}
            title={introContent.title}
            contents={introContent.contents}
            image={introContent.image}
            video={introContent.video}
            photoGrid={introContent.photoGrid}
          />
        ) : null}

        {hasSearchBar || hasFilters ? <BrowseTitle /> : null}

        {hasSearchBar ? <SearchBar onSearch={onDegreeSearch} /> : null}
        {hasFilters ? (
          <Filters
            value={stateFilters}
            onValueChange={setStateFilters}
            onApplyFilters={onDegreeApplyFilters}
            onCleanFilters={onDegreeCleanFilters}
          />
        ) : null}

        {error && <div>Something went wrong ...</div>}

        <section className="container m-1">
          <div className="d-flex justify-content-between">
            {hasFilters ? (
              <FiltersSummary
                appliedFilters={appliedFilters}
                onRemoveFilter={onRemoveFilterValue}
              />
            ) : null}

            {/* TODO: THIS COMPONENT IS CURRENTLY DEFERRED */}
            {/* <DataViewSwitch
              onChange={selectedViewId => {
                setSearchLoading(true);
                setDataViewComponent(selectedViewId);
                setSearchLoading(false);
              }}
              checkedId={dataViewComponent}
            /> */}
          </div>
        </section>

        {loading || searchLoading ? (
          <Loader />
        ) : (
          <ProgramList
            dataViewComponent={dataViewComponent}
            loading={loading || searchLoading}
            programms={tableView}
            columSettings={programList.settings}
          />
        )}
      </Main>
    </>
  );
};

ListingPage.propTypes = {
  actionUrls: PropTypes.shape({
    applyNowUrl: PropTypes.string,
  }),
  hasSearchBar: PropTypes.bool,
  hasFilters: PropTypes.bool,
  hero: PropTypes.shape(Hero.propTypes),
  introContent: PropTypes.shape(IntroContent.propTypes),
  programList: PropTypes.shape({
    dataSource: PropTypes.oneOfType([dataSourcePropType, PropTypes.string]),
    settings: columSettingsPropType,
  }),
};

export { ListingPage };
