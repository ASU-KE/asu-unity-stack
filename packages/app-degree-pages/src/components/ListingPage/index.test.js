// @ts-check
import React from "react";
import renderer from "react-test-renderer";

import { Button } from "../../../../components-core/src/components/Button";
// import { ListingPage } from "./index";

/**
 * @typedef {import('../../core/models/listing-page-types').ListingPageProps } AppProps
 */

/** @type {AppProps} */
const defaultArgs = {
  appPathFolder: ".", // OPTIONAL
  actionUrls: {
    applyNowUrl: "https://webapp4.asu.edu/uga_admissionsapp/?partner=CORP",
    majorInfoUrl: `?path=/story/program-detail-page--default&acadPlan={ACAD_PLAN_CODE}`,
  },
  hero: null,
  introContent: {
    type: "text",
    title: {
      text: "This is introductory marketing copy. Lorem ipsum dolor sit amet",
    },
    contents: [
      {
        text: `<p>Lorem, <strong>ipsum</strong></p>`,
      },
    ],
  },
  // hasFilters: false, // OPTIONAL
  // hasSearchBar: false, // OPTIONAL
  programList: {
    dataSource: {
      program: "graduate", // graduate | undergrad
      cert: "true", // "true" | "false"
    },
    settings: {
      hideCollegeSchool: true,
    },
  },
};

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <Button />
    // <ListingPage
    //   appPathFolder={defaultArgs.appPathFolder}
    //   actionUrls={defaultArgs.actionUrls}
    //   hero={defaultArgs.hero}
    //   introContent={defaultArgs.introContent}
    //   programList={defaultArgs.programList}
    //   hasSearchBar={defaultArgs.hasSearchBar}
    //   hasFilters={defaultArgs.hasFilters}
    // />
  );

  const page = component.toJSON();
  expect(page).toMatchSnapshot();
});
