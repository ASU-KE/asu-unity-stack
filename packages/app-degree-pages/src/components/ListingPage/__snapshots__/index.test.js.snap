// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="alert alert-danger"
  role="alert"
  style={
    Object {
      "display": "",
      "position": "sticky",
      "top": 0,
      "zIndex": 100,
    }
  }
>
  <div
    className="alert-icon"
  >
    <span
      aria-label="Error"
      className="fa fa-icon fa-exclamation-triangle"
    />
  </div>
  <div
    className="alert-content"
  >
    
  </div>
  <div
    className="alert-close"
  >
    <button
      aria-label="Close"
      className="btn btn-circle btn-circle-alt-black close"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-times"
      />
    </button>
  </div>
</div>
`;
